warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -240,32 +288,47 @@ public void reset() {
         quarantinedIds.clear();
     }
     
-    /*
-     * (non-Javadoc)
-     * 
-     * @see datawave.ingest.table.aggregator.PropogatingAggregator#propogateKey()
-     */
     @Override
     public boolean propogateKey() {
         
-        /**
-         * Changed logic so that if seenIgnore is true and count > MAX, we keep propogate the key
-         */
-        if ((seenIgnore && count > maxUids) || !quarantinedIds.isEmpty())
-            return true;
-        
-        HashSet<String> uidsCopy = new HashSet<>(uids);
-        uidsCopy.removeAll(uidsToRemove);
-        
-        if (log.isDebugEnabled()) {
-            log.debug(count + " " + uids.size() + " " + uidsToRemove.size() + " " + uidsCopy.size() + " removing " + (count == 0 && uidsCopy.isEmpty()));
+        // This method is called after reduce and then aggregate, so all of the work to combine has been done.
+        // If the propogate flag is true, then this might have been a partial major compaction, scan, or minor
+        // compaction and we want to keep all keys no matter what. When propogate is false, that means it's a full major
+        // compaction and therefore we can be certain that the aggregated result has combined all possible values for a
+        // given key. In that case, we only need to keep the resulting key/value pair if it has any UIDs (which means
+        // either UIDs in the quarantine or uid lists, or a positive uid count).
+        return propogate || !quarantinedIds.isEmpty() || !uids.isEmpty() || count > 0;
+    }
+    
+    @Override
+    public boolean validateOptions(Map<String,String> options) {
+        boolean valid = super.validateOptions(options);
+        if (valid) {
+            if (options.containsKey(TIMESTAMPS_IGNORED)) {
+                timestampsIgnored = Boolean.parseBoolean(options.get(TIMESTAMPS_IGNORED));
+            }
         }
-        
-        // if <= 0 and uids is empty, we can safely remove
-        if (count <= 0 && uidsCopy.isEmpty())
-            return false;
-        else
-            return true;
+        return valid;
     }
     
+    @Override
+    public SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironment env) {
+        GlobalIndexUidAggregator copy = (GlobalIndexUidAggregator) super.deepCopy(env);
+        copy.timestampsIgnored = timestampsIgnored;
+        copy.propogate = propogate;
+        // Not copying other fields that are all cleared in the reset() method.
+        return copy;
+    }
+    
+    @Override
+    public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
+        super.init(source, options, env);
+        if (options.containsKey(TIMESTAMPS_IGNORED)) {
```

Comment:
I hit an issue with trying to add options to GlobalIndexUidAggregator - since we're configuring it as an aggregator option on a TotalAggregatingIterator - the `GlobalIndexUidAggregator.init()` method is never actually invoked. Also, the TotalAggregatingIterator treats every option as a column so it'll treat `timestampsIgnored` as the column and try to instantiate a class of `true` for instance.

I also tried using GlobalIndexUidAggregator as a top-level iterator but there is an issue with that as the `propogateKey()` and `reset()` methods are invoked by TotalAggregatingIterator (via PropogatingIterator) - if you use GlobalIndexUidAggregator directly on the table, only `reduce()` gets invoked by Accumulo. 

I had considered maybe adding a hard-coded option to use as a flag to delegate options down into the aggregator but I ended up just using a subclass of GlobalIndexUidAggregator instead.

Reply:
@jschmidt10, PR #1091 would allow the options to be passed to aggregators via TotalAggregatingIterator. To use GlobalIndexUidAggregator as a Combiner, other work would probably be required.

Reply:
Great! Thanks.

Reply:
That PR also allows the max UID count to be configured. I can roll just the changes that allow the options to be passed along to this PR if that will make it easier to test anything. (Or, I can make that a separate PR.)

Reply:
I think we're set. With ignoreTimestamp being the default, we don't need the option right away.